packages/ui/src/elements/Autosave/index.tsx

```diff
@@ -88,6 +91,32 @@ export const Autosave: React.FC<Props> = ({ id, collection, global: globalDoc })
   // can always retrieve the most to date locale
   localeRef.current = locale
 
+  // Function to process the queue
+  const processQueue = useCallback(async () => {
+    if (isProcessing || queue.length === 0) {
+      return
+    }
+
+    setIsProcessing(true)
+    const currentTask = queue[0]
+
+    try {
+      await currentTask()
+    } catch (error) {
+      console.error('Error autosaving:', error)
+    } finally {
+      setQueue((prevQueue) => prevQueue.slice(1))
+      setIsProcessing(false)
+    }
+  }, [isProcessing, queue])
+
+  // Process the queue whenever it changes
+  useEffect(() => {
```

Comment:
Can we do this without a useEffect? Seems like it should be possible

Reply:
I would love to learn

Reply:
Maybe it cant, I hate it but will live

Reply:
Can't you just call `processCurrentAutosave` and have it take the autosave function as an arg and leave everything else as it is? Removing the autosave state and the useEffect?